src/main/java/com/github/tomakehurst/wiremock/extension/responsetemplating/ResponseTemplateModel.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2016-2025 Thomas Akehurst
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.tomakehurst.wiremock.extension.responsetemplating;
+
+import com.github.tomakehurst.wiremock.common.ListOrSingle;
+import com.github.tomakehurst.wiremock.http.Body;
+import java.util.Map;
+
+public class ResponseTemplateModel {
+
+  private final Body body;
+  private final int status;
+  private final Map<String, ListOrSingle<String>> headers;
+
+  public ResponseTemplateModel(int status, Map<String, ListOrSingle<String>> headers, Body body) {
+    this.body = body;
+    this.status = status;
+    this.headers = headers;
+  }
+
+  public int getStatus() {
+    return status;
+  }
+
+  public String getBody() {
+    return body.asBase64();
```

Comment:
I don't think this is what we want to expose to the templates.

The best approach would be to return the string representation when it's a textual body and null otherwise IMO.